src/electron.ts

```diff
@@ -90,6 +89,16 @@ app.on('activate', () => {
   }
 });
 
+function handleWindowsOnClose() {
+  unregisterShortcuts();
+  mainWindowState.saveState(current);
+  let index = windows.findIndex((w) => w.id === current.id);
+  windows.splice(index, 1);
+  current.close();
+  windows.filter(w => w);
```

Comment:
yes it was my obvious confusion that slice would do indexing like 0, 2, 3 etc. and I was missing filter assignment bas to array `windows = windows.filter(w => w);` Clear now.